flink-runtime/src/main/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManager.java

```diff
@@ -52,18 +61,45 @@
 public class ResourceManager extends RpcEndpoint<ResourceManagerGateway> {
 	private final ExecutionContext executionContext;
 	private final Map<JobMasterGateway, InstanceID> jobMasterGateways;
+	private final Map<ResourceID, TaskExecutorGateway> taskExecutorGateways;
+	private final Map<ResourceID, ResourceManagerToTaskExecutorHeartbeatScheduler> heartbeatSchedulers;
+	private final LeaderElectionService leaderElectionService;
+	private UUID leaderSessionID;
+	// TODO private final SlotManager slotManager;
 
-	public ResourceManager(RpcService rpcService, ExecutorService executorService) {
+	public ResourceManager(RpcService rpcService, ExecutorService executorService, LeaderElectionService leaderElectionService) {
 		super(rpcService);
 		this.executionContext = ExecutionContext$.MODULE$.fromExecutor(
 			Preconditions.checkNotNull(executorService));
 		this.jobMasterGateways = new HashMap<>();
+		this.taskExecutorGateways = new HashMap<>();
+		this.heartbeatSchedulers = new HashMap<>();
+		this.leaderElectionService = leaderElectionService;
+		leaderSessionID = null;
+		// TODO this.slotManager = null;
+	}
+
+	@Override
+	public void start() {
```

Comment:
It might actually makes sense to let the `start` method throw an exception so that a failed start call will result in a thrown exception.